sqlx-cli/src/opt.rs

```diff
@@ -121,6 +121,10 @@ pub enum MigrateCommand {
         /// else creates a single sql file
         #[clap(short)]
         reversible: bool,
+
+        /// If true, migrations are named sequentially instead of timestamp
+        #[clap(short)]
+        sequential: bool,
```

Comment:
I'd like us to figure out a better naming for this flag.

I hate unintuitively named short flags in CLIs and I feel like if I wanted sequential versioning I'd _constantly_ forget `-s`, but having to specify `--sequential` is also annoyingly verbose.

A coworker suggested `--inc` for "increment" but I don't really like that either. Maybe `-i`/`--increment`, but I don't think that's much better.

We could theoretically infer the style by looking at previous migrations, e.g.:

* Allow the user to override the behavior by supplying either `--timestamp` or `--sequential` (mutually exclusive).
* If there's at least two migrations, compare the last two: if their version numbers differ by 1, infer sequential.
    * This allows changing styles after the start of a project, which is something I've done before--using sequential
       for "baseline" migrations and timestamps for everything after the production database is created.
* If there's only one existing migration, infer sequential if the version number is 0 or 1.
    * I like using version 0 for a boilerplate/setup migration that gets copied from project to project.
* Otherwise, default to timestamp.

The more I think about inferring the style, the more I like it. It has the potential to cause the least friction because it errs on the side of consistency.

Reply:
@abonander thank you for detailed feedback!

I also like the idea of inferring the order based on the established one; I've implemented it in https://github.com/launchbadge/sqlx/pull/2602/commits/e275cd0b5cee57ac62ac8645fbf59225bef7d3bb — please take a look